src/main/scala/is/hail/expr/Relational.scala

```diff
@@ -1406,7 +1406,13 @@ case class MatrixMapCols(child: MatrixIR, newCol: IR, newKey: Option[IndexedSeq[
     val colValuesType = TArray(prev.typ.colType)
     val vaType = prev.typ.rvRowType
 
+    var initOpNeedsSA = false
+    var initOpNeedsGlobals = false
+
     val transformInitOp: (Int, IR) => IR = { (nAggs, initOpIR) =>
+      initOpNeedsSA = Mentions(initOpIR, "sa")
```

Comment:
I think I prefer if the  was returned from `CompileWithAggregators`. I'm thinking of something like
```
    val (rvAggs, initOps, seqOps, aggResultType, f, rTyp, mentions) = ir.CompileWithAggregators[Long, Long, Long, Long, Long, Long](
      "global", localGlobalsType,
      "sa", prev.typ.colType,
      "global", localGlobalsType,
      "colValues", colValuesType,
      "va", vaType,
      newCol,
      transformInitOp,
      transformSeqOp)
```
where `mentions` is a Map from any argument name to a boolean. Example: `Map("sa" -> true, "global" -> false)`.

Reply:
There'd have to be two maps.

Reply:
Ok, so. I very much agree what I did was an ugly hack.

I'm not a big fan of returning a mentions map since it seems oddly specific.

Instead I reworked CompileWithAggregators to provide the raw IRs back to the callee. What do you think of this set up?